llvm/lib/Transforms/Scalar/Sink.cpp

```diff
@@ -60,10 +63,62 @@ static bool isSafeToMove(Instruction *Inst, AliasAnalysis &AA,
   return true;
 }
 
+static cl::opt<unsigned> SinkLoadStoreLimit(
+    "sink-load-store-limit", cl::Hidden, cl::init(4),
+    cl::desc("Maximum number of stores in descendant blocks that will be "
+             "analyzed when attempting to sink a load."));
```

Comment:
Can you avoid this super specific flag? Is there some threshold MemorySSA usually uses? At worst can it be a pass parameter instead?

Reply:
I see a MaxCheckLimit in MemorySSA.cpp, but it is local to MemorySSA.cpp.  EarlyCSE.cpp, DeadStoreElimination.cpp, and LICM.cpp all have hidden options setting limits on memory SSA usage.  Is it okay if we stick with a hidden option?  Otherwise, by "pass parameter" do you mean a limit specific to Sink that is not implemented as a user option?